lib/vmdb/loggers.rb

```diff
@@ -76,6 +80,37 @@ def self.create_logger(log_file_name, logger_class = VMDBLogger)
       nil
     end
 
+    private_class_method def self.progname_from_file(log_file_name)
+      File.basename(log_file_name, ".*")
+    end
+
+    private_class_method def self.create_wrapper_logger(log_file_name, logger_class, wrapped_logger)
+      log_file_name = Pathname.new(log_file_name) unless log_file_name.kind_of?(Pathname)
+      progname      = progname_from_file(log_file_name)
+
+      logger_class.new(nil, progname: progname).tap do |logger|
+        # HACK: In order to access the "filename" of the wrapped logger, we inject it as an instance var.
+        logger.instance_variable_set(:@log_file_name, log_file_name)
+
+        def logger.filename
+          @log_file_name
```

Comment:
Ah okay you need it for `filename`

Reply:
Is this ok to be a path?

Also feels like we could avoid calling `filename` but maybe that is used by other code as well

Reply:
Yeah the problem is that other code currently calls `.filename`, so we need _something_ there.  However, the default `.filename` method goes through the log device, but since our log device is nil, then filename is also nil.  This overrides the default to just use the filename that's given (for now).